### ISSUE TITLE
Authentication header bypassing-authentication

### ISSUE THREAD

**Issue by batiati** [kubeapps/kubeapps#1839]
### Description:

I'm trying to bypass the login form injecting the authentication header on the NGINX ingress, following the instructions as stated on this doc:

https://github.com/kubeapps/kubeapps/blob/master/docs/user/login-alternatives.md#bypassing-authentication

![image](https://user-images.githubusercontent.com/5728002/86246304-28ecbc00-bb81-11ea-8605-da4600e0f3fe.png)

But it still showing the Login form requesting the token.
I'm not using any OAuth2/OIDC provider, I just wanted to publish an internal demo without the need of authenticate.

From the ingress log, it seems that the `add_header` directive needs an additional argument:

```
Error: exit status 1
2020/07/01 11:50:04 [emerg] 56664#56664: invalid number of arguments in "add_header" directive in /tmp/nginx-cfg745122493:760
nginx: [emerg] invalid number of arguments in "add_header" directive in /tmp/nginx-cfg745122493:760
nginx: configuration file /tmp/nginx-cfg745122493 test failed
```
I tried to fix it putting the [always parameter ](http://nginx.org/en/docs/http/ngx_http_headers_module.html#add_header), which solved the problem on the NGINX ingress log, but had no effect on kubeapps login form.

### Version of Helm, Kubeapps and Kubernetes:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.2.0", GitCommit:"e11b7ce3b12db2941e90399e874513fbd24bcb71", GitTreeState:"clean", GoVersion:"go1.13.10"}
```

- Output of `helm list <kubeapps-release-name>`:

```
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART           APP VERSION
kubeapps        kubeapps        2               2020-06-29 20:57:44.7007997 -0300 -03   deployed        kubeapps-3.7.2  v1.10.2
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.8", GitCommit:"ec6eb119b81be488b030e849b9e64fda4caaf33c", GitTreeState:"clean", BuildDate:"2020-03-12T21:00:06Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.4", GitCommit:"8d8aa39598534325ad77120c120a22b3a990b5ea", GitTreeState:"clean", BuildDate:"2020-03-12T20:55:23Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by andresmgot** [comment]
Hi @batiati,

It seems that those instructions are a bit out of date. Honestly, I don't know if that's something we should support (copying a token should be easy enough for a demo?). In any case, I've given it a try. What I did was to modify the Kubeapps config (`kubectl edit -n kubeapps configmap kubeapps-internal-dashboard-config`) and enable there as if I were using the `authProxy` (`"authProxyEnabled": true,`). Then, in the Ingress configuration, rather than using `add_header` I used `proxy_set_header`. This allowed me to bypass the login page. I found an error when listing apps but that may be because of my development environment. I could see the catalog.

Let us know if that works for you,
